edsnlp/pipes/llm/llm_span_qualifier/llm_span_qualifier.py

```diff
@@ -0,0 +1,794 @@
+    Type,
+    Union,
+)
+
+from pydantic import BaseModel
+from spacy.tokens import Doc, Span
+from typing_extensions import Annotated, Literal
+
+from edsnlp.core import PipelineProtocol
+from edsnlp.pipes.base import BaseSpanAttributeClassifierComponent
+from edsnlp.utils.bindings import BINDING_GETTERS, BINDING_SETTERS, AttributesArg
+from edsnlp.utils.span_getters import ContextWindow, SpanGetterArg, get_spans
+
+from ..async_worker import AsyncRequestWorker
+
+
+class LlmSpanQualifier(BaseSpanAttributeClassifierComponent):
+    r'''
+    The `eds.llm_span_qualifier` component qualifies spans using a
+    Large Language Model (LLM) that returns structured JSON attributes.
+
+    This component takes existing spans, wraps them with `<ent>` markers inside a
+    context window and prompts an LLM to answer with a JSON object that matches the
+    configured schema. The response is validated and written back on the span
+    extensions.
+
+    In practice, along with a system prompt that constrains the allowed attributes
+    and optional few-shot examples provided as previous user / assistant messages,
+    the component sends snippets such as:
+    ```
+    Biopsies du <date>12/02/2025</date> : adénocarcinome.
+    ```
+
+    and expects a minimal JSON answer, for example:
+    ```json
+    {"biopsy_procedure": "yes"}
+    ```
+    which is then parsed and assigned to the span attributes.
+
+
+    !!! warning "Experimental"
+
+        This component is experimental. The API and behavior may change in future
+        versions. Make sure to pin your `edsnlp` version if you use it in a project.
+
+    !!! note "Dependencies"
+
+        This component requires several dependencies. Run the following command to
+        install them:
+        ```bash { data-md-color-scheme="slate" }
+        pip install openai bm25s Stemmer
+        ```
+        We recommend even to add them to your `pyproject.toml` or `requirements.txt`.
+
+    Examples
+    --------
+    If your data is sensitive, we recommend you to use a self-hosted
+    model with an OpenAI-compatible API, such as
+    [vLLM](https://github.com/vllm-project/vllm).
+
+    You can store your OpenAI API key in the `OPENAI_API_KEY` environment
+    variable.
+    ```python { .no-check }
+    import os
+    os.environ["OPENAI_API_KEY"] = "your_api_key_here"
+    ```
+
+    Start a server with the model of your choice:
+
+    ```bash { data-md-color-scheme="slate" }
+    python -m vllm.entrypoints.openai.api_server \
+       --model mistral-small-24b-instruct-2501 \
+       --port 8080 \
+       --enable-prefix-caching
+    ```
+
+    You can then use the `llm_span_qualifier` component as follows:
+
+    <!-- blacken-docs:off -->
+
+    === "Yes/no bool classification"
+
+        ```python { .no-check }
+        from typing import Annotated
+        from pydantic import BeforeValidator, PlainSerializer, WithJsonSchema
+        import edsnlp, edsnlp.pipes as eds
+
+        BiopsySchema = Annotated[
```

Comment:
@percevalw  we should clarify this example as discussed. 
Maybe also inverse the order of examples, the more basic example in first place